hermes/probe/create_file.go

```diff
@@ -0,0 +1,124 @@
+package probe
+
+import (
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"math/rand"
+	// "os"
+	// "sync"
+
+	// "cloud.google.com/go/storage"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	// "github.com/google/cloudprober/logger"
+	// "github.com/google/cloudprober/metrics"
+	// "github.com/google/cloudprober/probes/options"
+	// "github.com/google/cloudprober/targets/endpoint"
+	// "github.com/googleinterns/step224-2020/blob/master/hermes/hermes.go"
+)
+
+const (
+	begin = 1  // ID of the first HermesFile
+	end   = 50 // ID of the last HermesFile
+	maxFileSize = 1000 // maximum allowed file size in bytes 
+)
+
+type RandomHermesFile struct {
+	ID   int64 // ID is a positive integer in the range [1,50]
+	Size int   // File size in bytes
+}
+
+type randomHermesFileReader struct {
+	size int // size in bytes
+	i    int // currently reading this byte
+	rand *rand.Rand
+}
+
+func (r *randomHermesFileReader) readDone() bool {
+	return r.i >= r.size
+}
+
+func (r *randomHermesFileReader) bytesLeft() int {
+	return (r.size - r.i)
+}
+
+func (r *randomHermesFileReader) bufferTooLong(buffer []byte) bool {
+	return len(buffer) > r.bytesLeft()
+}
+
+func (r *randomHermesFileReader) Read(buff []byte) (n int, err error) {
+	if r.readDone() {
+		return 0, io.EOF
+	}
+	b := buff
+	if r.bufferTooLong(b){
+		b = buff[:r.bytesLeft()]
+	}
+	n, err = r.rand.Read(b)
+	if(err!=nil){
+		return 0, err // 0 as nil casn't be returned as a type int argument
+	}
+	r.i += n
+	return
+}
+
+// Warning: NewReader is not thread safe
+func (f *RandomHermesFile) NewReader() *randomHermesFileReader {
+	return &randomHermesFileReader{size: f.Size, rand: rand.New(rand.NewSource(f.ID))} // ID will serve as a Seed and i will be set to 0 automatically
+}
+
+func (f *RandomHermesFile) CheckSum() ([]byte, error) {
+	r := f.NewReader()
+	h := sha1.New()
+	if _, err := io.Copy(h, r); err != nil {
+		return nil, fmt.Errorf("io.Copy: %v", err)
+	}
+	// returns checksum in hex notation
+	return h.Sum(nil), nil
+}
+
+func (f *RandomHermesFile) FileName() (string, error) {
+	if f.ID < begin || f.ID > end {
+		return "", fmt.Errorf("The file ID provided %v wasn't in the required range [1,50]", f.ID)
```

Comment:
Nit: There are consts for the lower and upper bound. Use those instead of hard-coding the values in the string.
Otherwise it's very hard to keep the in sync.

Reply:
changed